src/postgres/commands/createPostgresServer/steps/PostgresServerNameStep.ts

```diff
@@ -46,13 +47,10 @@ async function validatePostgresServerName(name: string, client: PostgreSQLManage
         return localize('serverNamePrefixSuffixCheck', 'Server name must not start or end in a hyphen.');
     }
 
-    const availabilityRequest: NameAvailabilityRequest = { name: name, type: "Microsoft.DBforPostgreSQL" };
-    const availability: NameAvailability = (await client.checkNameAvailability.execute(availabilityRequest));
+    const availability: AbstractNameAvailability = await client.checkNameAvailability.execute({name: name, type: "Microsoft.DBforPostgreSQL"});
 
     if (!availability.nameAvailable) {
-        if (availability.reason === 'AlreadyExists') {
-            return localize('serverNameAvailabilityCheck', 'A server named "{0}" already exists.', name);
-        }
+        return localize('serverNameAvailabilityCheck', 'Server name "{0}" is not available.', name);
```

Comment:
Let's just return `availability.message` - not sure why we weren't using that already